### DISCUSSION TITLE
A way for interface extension

### DISCUSSION THREAD

**Discussion by alvin4712** [mapstruct/mapstruct#3379] [General]
Is there any way for the mapper to inherit from a generic interface so that all mappers have the same signature and that the input and output are defined by parameters.

```java
interface mapinterface <MDL, DTO> {
         DTO toDto(MDL modelo);
         MDL toModel(DTO dto);
}
```

```java
@Mapper
interface CarMapper extends mapinterface< Car, CarDto> {}
```

**Comment by filiphr** [comment]
Have you tried that out? You can extend any interface you want in your own mappers. In your example MapStruct is properly going to create an implementation of the `toDto` and `toModel` methods
